prow/cmd/peribolos/main.go

```diff
@@ -262,12 +292,101 @@ func dumpOrgConfig(client dumpClient, orgName string) (*org.Config, error) {
 		return t
 	}
 
-	out.Teams = make(map[string]org.Team, len(tops))
+	result := make(map[string]org.Team, len(tops))
 	for _, id := range tops {
-		out.Teams[names[id]] = makeChild(id)
+		result[names[id]] = makeChild(id)
 	}
 
-	return &out, nil
+	return result, nil
+}
+
+func createRepoConfig(client dumpClient, orgName, repoList string, members []string) (map[string]org.Repo, error) {
+	repos := strings.Split(repoList, ",")
+	if len(repos) == 0 {
+		// get all repos
+		allRepos, err := client.GetRepos(orgName, false)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get repos: %v", err)
+		}
+
+		for _, r := range allRepos {
+			repos = append(repos, r.Name)
+		}
+	}
+
+	result := make(map[string]org.Repo, len(repos))
+	for _, repo := range repos {
+		// get outside collaborators
+		oc, err := client.ListCollaborators(orgName, repo, true)
+		if err != nil {
+			return nil, err
+		}
+		var ocAdmin, ocWrite, ocRead []string
+		for _, user := range oc {
+			permission, err := client.GetUserPermission(orgName, repo, user.Login)
+			if err != nil {
+				return nil, err
+			}
+			switch permission {
+			case string(org.Admin):
+				ocAdmin = append(ocAdmin, user.Login)
+			case string(org.Write):
+				ocWrite = append(ocWrite, user.Login)
+			case string(org.Read):
+				ocRead = append(ocRead, user.Login)
+			}
+		}
+
+		// get members with admin and write access
+		// don't consider read access for now because all members have implicit read access
+		var membersAdmin, membersWrite []string
+		for _, user := range members {
+			permission, err := client.GetUserPermission(orgName, repo, user)
```

Comment:
I'm concerned this could lead to an increased token usage. Thoughts?

Reply:
I suspect our caching will make this irrelevant anyway (since this is a GET request)